diff --git a/angle/src/libANGLE/renderer/metal/ProgramExecutableMtl.mm b/angle/src/libANGLE/renderer/metal/ProgramExecutableMtl.mm
index 04fc45023..b2a4f3f96 100644
--- a/angle/src/libANGLE/renderer/metal/ProgramExecutableMtl.mm
+++ b/angle/src/libANGLE/renderer/metal/ProgramExecutableMtl.mm
@@ -248,7 +248,7 @@ void InitArgumentBufferEncoder(mtl::Context *context,
     if (encoder->metalArgBufferEncoder)
     {
         encoder->bufferPool.initialize(context, encoder->metalArgBufferEncoder.get().encodedLength,
-                                       mtl::kArgumentBufferOffsetAlignment, 0);
+                                       mtl::kArgumentBufferOffsetAlignment, 10);
     }
 }
 
diff --git a/angle/src/libANGLE/renderer/metal/ProvokingVertexHelper.mm b/angle/src/libANGLE/renderer/metal/ProvokingVertexHelper.mm
index 3c3b47be1..0a186b63a 100644
--- a/angle/src/libANGLE/renderer/metal/ProvokingVertexHelper.mm
+++ b/angle/src/libANGLE/renderer/metal/ProvokingVertexHelper.mm
@@ -101,7 +101,7 @@ static inline gl::PrimitiveMode getNewPrimitiveMode(const uint fixIndexBufferKey
 }
 ProvokingVertexHelper::ProvokingVertexHelper(ContextMtl *context) : mIndexBuffers(false)
 {
-    mIndexBuffers.initialize(context, kInitialIndexBufferSize, mtl::kIndexBufferOffsetAlignment, 0);
+    mIndexBuffers.initialize(context, kInitialIndexBufferSize, mtl::kIndexBufferOffsetAlignment, 10);
 }
 
 void ProvokingVertexHelper::onDestroy(ContextMtl *context)
diff --git a/angle/src/libANGLE/renderer/metal/VertexArrayMtl.mm b/angle/src/libANGLE/renderer/metal/VertexArrayMtl.mm
index 1d30b2534..50e894ac3 100644
--- a/angle/src/libANGLE/renderer/metal/VertexArrayMtl.mm
+++ b/angle/src/libANGLE/renderer/metal/VertexArrayMtl.mm
@@ -198,7 +198,7 @@ VertexArrayMtl::VertexArrayMtl(const gl::VertexArrayState &state, ContextMtl *co
                                   /** maxBuffers */ 10 * mtl::kMaxVertexAttribs);
 
     mDynamicIndexData.initialize(context, kDynamicIndexDataSize, mtl::kIndexBufferOffsetAlignment,
-                                 0);
+                                 10);
 }
 VertexArrayMtl::~VertexArrayMtl() {}
 
